api/v01/entities/vrp_input.rb

```diff
@@ -516,11 +516,15 @@ module VrpVehicles
     optional(:distance, type: Integer, desc: 'Maximum tour distance. Not available with periodic heuristic.')
     optional(:maximum_ride_time, type: Integer, desc: 'Maximum ride duration between two route activities')
     optional(:maximum_ride_distance, type: Integer, desc: 'Maximum ride distance between two route activities')
-    optional :skills, type: Array[Array[Symbol]],
+    optional :skills, type: Array[Array[Symbol]], default: [[]],
                       coerce_with: ->(val) {
-                        val.is_a?(String) ?
-                        [val.split(/,/).map!(&:strip).map!(&:to_sym)] :
-                        val&.map{ |set| set.map(&:to_sym) }
+                        if val.is_a?(String)
+                          [val.split(/,/).map!(&:strip).map!(&:to_sym)]
+                        elsif val == [] # Mapotempo-Web case
+                          [[]]
+                        else
+                          val&.map{ |set| set.map(&:to_sym) }
+                        end
```

Comment:
coerce_with is called for each "element" of the array, so returning `[[]]` is not correct.

Reply:
If mapotempo-web is sending `[]` then this coerce_with is not called (because the array is empty -- i.e., there is no `val` to call the coerce with).

Reply:
When mapotempo-web sends [ ] then val is [] and we do call coerce on [], otherwise I guess test_generated_vehicle_skills would fail.

Reply:
Sorry, I just verified that this is the case with a byebug, I mixed "values" logic with the "coerce_with" logic -- one is called for each value of the array, the other is called for the value as whole :facepalm:

Reply:
What would this coerce do for `["skill1", "skill2"]` instead of `[["skill1", "skill2"]]` (I don't know if mt-web can send such skills)?

Reply:
I don't know either, @braktar do you know ?
Since that wasn't in vehicle.rb create I just ignored this case, assuming it was already managed.

Reply:
In mt-web, vehicle skills are transmitted the following way : `data[:use_skills] ? [vehicle[:skills]] : nil`

Reply:
I am not so sure that it is managed. I think, if it is okay for you, we can fix this coerce_with so that it handles all cases correctly -- so that we are sure that only correct vehicle[:skills] enters the API and all coerce logic is in one place.

Reply:
Thanks @braktar, I don't see how they can send us `[]` in that case? When the vehicle has no skills, they should be sending `[[]]` or `[nil]` (a la marge).